# GPTQ + Marlin 최적화 버전

## 개요
vLLM의 Marlin 커널에 최적화된 GPTQ 양자화입니다.

### Marlin 커널이란?
- NVIDIA GPU용 초고속 4-bit 추론 커널
- 일반 GPTQ 대비 2~4배 빠름
- vLLM이 자동으로 적용

### Marlin 호환 조건
- W4A16 (4-bit weights, 16-bit activations)
- group_size = 128
- 표준 GPTQ 포맷

---

# 1. Import 및 환경 확인

In [1]:
import os
import sys
import torch
import shutil
from pathlib import Path

from datasets import load_dataset
from transformers import AutoModelForCausalLM, AutoTokenizer

from llmcompressor import oneshot
from llmcompressor.modifiers.quantization import GPTQModifier

print("=" * 60)
print("환경 정보")
print("=" * 60)
print(f"Python: {sys.version.split()[0]}")
print(f"PyTorch: {torch.__version__}")
print(f"CUDA 사용 가능: {torch.cuda.is_available()}")
if torch.cuda.is_available():
    print(f"GPU: {torch.cuda.get_device_name(0)}")
else:
    print("⚠️  GPU 없음 - CPU로 실행됩니다")
print("=" * 60)

환경 정보
Python: 3.10.13
PyTorch: 2.9.1
CUDA 사용 가능: False
⚠️  GPU 없음 - CPU로 실행됩니다


# 2. 하이퍼파라미터 설정

### Marlin 최적화 포인트
1. `group_size=128` (Marlin 필수)
2. 높은 캘리브레이션 품질 (성능 유지)
3. `actorder="weight"` (정확도 향상)

In [2]:
# ============================================================================
# 모델 설정
# ============================================================================
# 로컬 모델 경로 (다운로드 불필요!)
MODEL_ID = "./open/base_model"
OUT_DIR = "./model"  # 제출용 폴더명

# 데이터셋 설정
DATASET_ID = "LGAI-EXAONE/MANTA-1M"
DATASET_SPLIT = "train"

# ============================================================================
# 캘리브레이션 설정 (품질 우선)
# ============================================================================
# Marlin은 속도를 올려주므로, 성능(PerfNorm) 유지가 핵심!
if torch.cuda.is_available():
    NUM_CALIBRATION_SAMPLES = 512
    MAX_SEQUENCE_LENGTH = 1024
else:
    NUM_CALIBRATION_SAMPLES = 256  # CPU에서도 품질 유지
    MAX_SEQUENCE_LENGTH = 512

# ============================================================================
# Marlin 호환 GPTQ 설정
# ============================================================================
SCHEME = "W4A16"          # 4-bit weights, 16-bit activations
GROUP_SIZE = 128          # Marlin 필수 조건!
ACTORDER = "weight"       # 정확도 향상 (weight 기반 순서)
DAMPENING_FRAC = 0.01     # Hessian 안정화

TARGETS = ["Linear"]
IGNORE = ["embed_tokens", "lm_head"]  # 임베딩 레이어 제외

# 원본 모델 크기
ORIGINAL_MODEL_SIZE_GB = 2.56

print("=" * 60)
print("GPTQ + Marlin 최적화 설정")
print("=" * 60)
print(f"MODEL_ID: {MODEL_ID}")
print(f"OUT_DIR: {OUT_DIR}")
print(f"SCHEME: {SCHEME}")
print(f"GROUP_SIZE: {GROUP_SIZE} (Marlin 호환)")
print(f"ACTORDER: {ACTORDER}")
print(f"NUM_CALIBRATION_SAMPLES: {NUM_CALIBRATION_SAMPLES}")
print(f"MAX_SEQUENCE_LENGTH: {MAX_SEQUENCE_LENGTH}")
print("=" * 60)

GPTQ + Marlin 최적화 설정
MODEL_ID: ./open/base_model
OUT_DIR: ./model
SCHEME: W4A16
GROUP_SIZE: 128 (Marlin 호환)
ACTORDER: weight
NUM_CALIBRATION_SAMPLES: 256
MAX_SEQUENCE_LENGTH: 512


# 3. 모델 로드

In [3]:
print("[INFO] 모델 로드 중...")

tokenizer = AutoTokenizer.from_pretrained(
    MODEL_ID,
    trust_remote_code=True,
)

device_map = "auto" if torch.cuda.is_available() else "cpu"

model = AutoModelForCausalLM.from_pretrained(
    MODEL_ID,
    torch_dtype=torch.float32 if not torch.cuda.is_available() else torch.bfloat16,
    trust_remote_code=True,
    device_map=device_map,
)

print(f"[INFO] 모델 파라미터: {model.num_parameters():,}")
print(f"[INFO] 디바이스: {device_map}")
print("[INFO] 모델 로드 완료")

`torch_dtype` is deprecated! Use `dtype` instead!


[INFO] 모델 로드 중...
[INFO] 모델 파라미터: 1,279,391,488
[INFO] 디바이스: cpu
[INFO] 모델 로드 완료


# 4. 캘리브레이션 데이터셋

In [4]:
print("[INFO] 캘리브레이션 데이터 로드 중...")

ds = load_dataset(
    DATASET_ID,
    split=f"{DATASET_SPLIT}[:{NUM_CALIBRATION_SAMPLES}]",
)

def preprocess(example):
    return {
        "text": tokenizer.apply_chat_template(
            example["conversations"],
            add_generation_prompt=True,
            tokenize=False
        )
    }

ds = ds.map(preprocess)

print(f"[INFO] 데이터셋 크기: {len(ds)}")
print("[INFO] 데이터 전처리 완료")

[INFO] 캘리브레이션 데이터 로드 중...
[INFO] 데이터셋 크기: 256
[INFO] 데이터 전처리 완료


# 5. GPTQ 양자화 (Marlin 호환)

In [5]:
print("[INFO] GPTQ 양자화 시작 (Marlin 호환)")
print(f"  - scheme: {SCHEME}")
print(f"  - group_size: {GROUP_SIZE} (Marlin 필수)")
print(f"  - actorder: {ACTORDER}")
print(f"  - samples: {NUM_CALIBRATION_SAMPLES}")

if torch.cuda.is_available():
    print("\n🚀 GPU 모드: 15-30분 예상\n")
else:
    print("\n⏳ CPU 모드: 2-4시간 예상\n")

# ============================================================================
# Marlin 호환 GPTQ 레시피
# ============================================================================
recipe = [
    GPTQModifier(
        scheme=SCHEME,
        targets=TARGETS,
        ignore=IGNORE,
        block_size=GROUP_SIZE,      # Marlin 호환: 128
        dampening_frac=DAMPENING_FRAC,
        actorder=ACTORDER,          # weight 기반 정렬
    )
]

oneshot(
    model=model,
    dataset=ds,
    recipe=recipe,
    max_seq_length=MAX_SEQUENCE_LENGTH,
    num_calibration_samples=NUM_CALIBRATION_SAMPLES,
)

print("\n[INFO] GPTQ 양자화 완료!")
print("[INFO] vLLM에서 자동으로 Marlin 커널 적용됨")

[INFO] GPTQ 양자화 시작 (Marlin 호환)
  - scheme: W4A16
  - group_size: 128 (Marlin 필수)
  - actorder: weight
  - samples: 256

⏳ CPU 모드: 2-4시간 예상



Tokenizing:   0%|          | 0/256 [00:00<?, ? examples/s]

2026-02-09T22:41:14.225875+0900 | reset | INFO - Compression lifecycle reset
2026-02-09T22:41:14.226989+0900 | from_modifiers | INFO - Creating recipe from modifiers
2026-02-09T22:41:14.245392+0900 | initialize | INFO - Compression lifecycle initialized for 1 modifiers
2026-02-09T22:41:14.245755+0900 | IndependentPipeline | INFO - Inferred `SequentialPipeline` for `GPTQModifier`
2026-02-09T22:41:14.250961+0900 | dispatch_for_sequential | WARNING - CUDA/XPU is not available! Compressing model on CPU instead


W0209 22:41:14.272000 71243 torch/fx/_symbolic_trace.py:52] is_fx_tracing will return true for both fx.symbolic_trace and torch.export. Please use is_fx_tracing_symbolic_tracing() for specifically fx.symbolic_trace or torch.compiler.is_compiling() for specifically torch.export/compile.
(1/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:36<00:00,  6.98it/s]

2026-02-09T22:41:51.118714+0900 | compress_modules | INFO - Quantizing model.layers.0.self_attn.q_proj using 256 samples


2026-02-09T22:41:51.537685+0900 | compress | METRIC - time 0.42s
2026-02-09T22:41:51.538090+0900 | compress | METRIC - error 1.12
2026-02-09T22:41:51.539050+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:41:51.539315+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:41:51.540941+0900 | compress_modules | INFO - Quantizing model.layers.0.self_attn.k_proj using 256 samples
2026-02-09T22:41:51.727360+0900 | compress | METRIC - time 0.19s
2026-02-09T22:41:51.727737+0900 | compress | METRIC - error 0.33
2026-02-09T22:41:51.728568+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:41:51.728802+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:41:51.729497+0900 | compress_modules | INFO - Quantizing model.layers.0.self_attn.v_proj using 256 samples
2026-02-09T22:41:51.915684+0900 | compress | METRIC - time 0.19s
2026-02-09T22:41:51.91

(2/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:40<00:00,  6.39it/s]

2026-02-09T22:42:42.758927+0900 | compress_modules | INFO - Quantizing model.layers.1.self_attn.q_proj using 256 samples


2026-02-09T22:42:43.235361+0900 | compress | METRIC - time 0.48s
2026-02-09T22:42:43.235780+0900 | compress | METRIC - error 4.77
2026-02-09T22:42:43.237087+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:42:43.237410+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:42:43.239011+0900 | compress_modules | INFO - Quantizing model.layers.1.self_attn.k_proj using 256 samples
2026-02-09T22:42:43.586281+0900 | compress | METRIC - time 0.35s
2026-02-09T22:42:43.586699+0900 | compress | METRIC - error 1.36
2026-02-09T22:42:43.587562+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:42:43.587776+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:42:43.588365+0900 | compress_modules | INFO - Quantizing model.layers.1.self_attn.v_proj using 256 samples
2026-02-09T22:42:43.873474+0900 | compress | METRIC - time 0.28s
2026-02-09T22:42:43.87

(3/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:39<00:00,  6.53it/s]

2026-02-09T22:43:34.868851+0900 | compress_modules | INFO - Quantizing model.layers.2.self_attn.q_proj using 256 samples


2026-02-09T22:43:35.162090+0900 | compress | METRIC - time 0.29s
2026-02-09T22:43:35.162433+0900 | compress | METRIC - error 12.96
2026-02-09T22:43:35.163304+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:43:35.163523+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:43:35.164828+0900 | compress_modules | INFO - Quantizing model.layers.2.self_attn.k_proj using 256 samples
2026-02-09T22:43:35.347546+0900 | compress | METRIC - time 0.18s
2026-02-09T22:43:35.347882+0900 | compress | METRIC - error 3.64
2026-02-09T22:43:35.348682+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:43:35.348890+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:43:35.349464+0900 | compress_modules | INFO - Quantizing model.layers.2.self_attn.v_proj using 256 samples
2026-02-09T22:43:35.530177+0900 | compress | METRIC - time 0.18s
2026-02-09T22:43:35.5

(4/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:39<00:00,  6.56it/s]

2026-02-09T22:44:25.045448+0900 | compress_modules | INFO - Quantizing model.layers.3.self_attn.q_proj using 256 samples


2026-02-09T22:44:25.346742+0900 | compress | METRIC - time 0.30s
2026-02-09T22:44:25.347166+0900 | compress | METRIC - error 26.44
2026-02-09T22:44:25.348035+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:44:25.348294+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:44:25.349805+0900 | compress_modules | INFO - Quantizing model.layers.3.self_attn.k_proj using 256 samples
2026-02-09T22:44:25.545495+0900 | compress | METRIC - time 0.20s
2026-02-09T22:44:25.545983+0900 | compress | METRIC - error 7.47
2026-02-09T22:44:25.546787+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:44:25.547016+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:44:25.547717+0900 | compress_modules | INFO - Quantizing model.layers.3.self_attn.v_proj using 256 samples
2026-02-09T22:44:25.768264+0900 | compress | METRIC - time 0.22s
2026-02-09T22:44:25.7

(5/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:38<00:00,  6.70it/s]

2026-02-09T22:45:14.791326+0900 | compress_modules | INFO - Quantizing model.layers.4.self_attn.q_proj using 256 samples


2026-02-09T22:45:15.081983+0900 | compress | METRIC - time 0.29s
2026-02-09T22:45:15.082349+0900 | compress | METRIC - error 50.32
2026-02-09T22:45:15.083206+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:45:15.083451+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:45:15.084825+0900 | compress_modules | INFO - Quantizing model.layers.4.self_attn.k_proj using 256 samples
2026-02-09T22:45:15.265802+0900 | compress | METRIC - time 0.18s
2026-02-09T22:45:15.266151+0900 | compress | METRIC - error 13.93
2026-02-09T22:45:15.266977+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:45:15.267170+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:45:15.267853+0900 | compress_modules | INFO - Quantizing model.layers.4.self_attn.v_proj using 256 samples
2026-02-09T22:45:15.473111+0900 | compress | METRIC - time 0.21s
2026-02-09T22:45:15.

(6/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.82it/s]

2026-02-09T22:46:03.683207+0900 | compress_modules | INFO - Quantizing model.layers.5.self_attn.q_proj using 256 samples


2026-02-09T22:46:03.972946+0900 | compress | METRIC - time 0.29s
2026-02-09T22:46:03.973319+0900 | compress | METRIC - error 81.40
2026-02-09T22:46:03.974163+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:46:03.974405+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:46:03.975821+0900 | compress_modules | INFO - Quantizing model.layers.5.self_attn.k_proj using 256 samples
2026-02-09T22:46:04.159127+0900 | compress | METRIC - time 0.18s
2026-02-09T22:46:04.159475+0900 | compress | METRIC - error 23.90
2026-02-09T22:46:04.160306+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:46:04.160521+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:46:04.161194+0900 | compress_modules | INFO - Quantizing model.layers.5.self_attn.v_proj using 256 samples
2026-02-09T22:46:04.343134+0900 | compress | METRIC - time 0.18s
2026-02-09T22:46:04.

(7/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.74it/s]

2026-02-09T22:46:52.604804+0900 | compress_modules | INFO - Quantizing model.layers.6.self_attn.q_proj using 256 samples


2026-02-09T22:46:52.901398+0900 | compress | METRIC - time 0.30s
2026-02-09T22:46:52.901786+0900 | compress | METRIC - error 118.11
2026-02-09T22:46:52.902680+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:46:52.902901+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:46:52.904335+0900 | compress_modules | INFO - Quantizing model.layers.6.self_attn.k_proj using 256 samples
2026-02-09T22:46:53.095707+0900 | compress | METRIC - time 0.19s
2026-02-09T22:46:53.096097+0900 | compress | METRIC - error 32.48
2026-02-09T22:46:53.096945+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:46:53.097193+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:46:53.097856+0900 | compress_modules | INFO - Quantizing model.layers.6.self_attn.v_proj using 256 samples
2026-02-09T22:46:53.293048+0900 | compress | METRIC - time 0.19s
2026-02-09T22:46:53

(8/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.79it/s]

2026-02-09T22:47:41.527434+0900 | compress_modules | INFO - Quantizing model.layers.7.self_attn.q_proj using 256 samples


2026-02-09T22:47:41.817776+0900 | compress | METRIC - time 0.29s
2026-02-09T22:47:41.818147+0900 | compress | METRIC - error 178.31
2026-02-09T22:47:41.818968+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:47:41.819192+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:47:41.820617+0900 | compress_modules | INFO - Quantizing model.layers.7.self_attn.k_proj using 256 samples
2026-02-09T22:47:42.003200+0900 | compress | METRIC - time 0.18s
2026-02-09T22:47:42.003571+0900 | compress | METRIC - error 50.15
2026-02-09T22:47:42.004368+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:47:42.004591+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:47:42.005162+0900 | compress_modules | INFO - Quantizing model.layers.7.self_attn.v_proj using 256 samples
2026-02-09T22:47:42.189251+0900 | compress | METRIC - time 0.18s
2026-02-09T22:47:42

(9/31): Calibrating: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.88it/s]

2026-02-09T22:48:29.719417+0900 | compress_modules | INFO - Quantizing model.layers.8.self_attn.q_proj using 256 samples


2026-02-09T22:48:30.007840+0900 | compress | METRIC - time 0.29s
2026-02-09T22:48:30.008195+0900 | compress | METRIC - error 195.12
2026-02-09T22:48:30.009009+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:48:30.009226+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:48:30.010583+0900 | compress_modules | INFO - Quantizing model.layers.8.self_attn.k_proj using 256 samples
2026-02-09T22:48:30.192614+0900 | compress | METRIC - time 0.18s
2026-02-09T22:48:30.192962+0900 | compress | METRIC - error 55.68
2026-02-09T22:48:30.193773+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:48:30.193986+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:48:30.194591+0900 | compress_modules | INFO - Quantizing model.layers.8.self_attn.v_proj using 256 samples
2026-02-09T22:48:30.378700+0900 | compress | METRIC - time 0.18s
2026-02-09T22:48:30

(10/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.87it/s]

2026-02-09T22:49:17.989846+0900 | compress_modules | INFO - Quantizing model.layers.9.self_attn.q_proj using 256 samples


2026-02-09T22:49:18.330604+0900 | compress | METRIC - time 0.34s
2026-02-09T22:49:18.330973+0900 | compress | METRIC - error 260.15
2026-02-09T22:49:18.331837+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:49:18.332065+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:49:18.333391+0900 | compress_modules | INFO - Quantizing model.layers.9.self_attn.k_proj using 256 samples
2026-02-09T22:49:18.518671+0900 | compress | METRIC - time 0.19s
2026-02-09T22:49:18.519011+0900 | compress | METRIC - error 76.72
2026-02-09T22:49:18.519811+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:49:18.520028+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:49:18.520723+0900 | compress_modules | INFO - Quantizing model.layers.9.self_attn.v_proj using 256 samples
2026-02-09T22:49:18.703891+0900 | compress | METRIC - time 0.18s
2026-02-09T22:49:18

(11/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.87it/s]

2026-02-09T22:50:06.299921+0900 | compress_modules | INFO - Quantizing model.layers.10.self_attn.q_proj using 256 samples


2026-02-09T22:50:06.593217+0900 | compress | METRIC - time 0.29s
2026-02-09T22:50:06.593581+0900 | compress | METRIC - error 282.98
2026-02-09T22:50:06.594410+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:50:06.594628+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:50:06.595977+0900 | compress_modules | INFO - Quantizing model.layers.10.self_attn.k_proj using 256 samples
2026-02-09T22:50:06.786548+0900 | compress | METRIC - time 0.19s
2026-02-09T22:50:06.786894+0900 | compress | METRIC - error 76.06
2026-02-09T22:50:06.787723+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:50:06.787948+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:50:06.788604+0900 | compress_modules | INFO - Quantizing model.layers.10.self_attn.v_proj using 256 samples
2026-02-09T22:50:06.971507+0900 | compress | METRIC - time 0.18s
2026-02-09T22:50:

(12/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.85it/s]

2026-02-09T22:50:54.719506+0900 | compress_modules | INFO - Quantizing model.layers.11.self_attn.q_proj using 256 samples


2026-02-09T22:50:55.008886+0900 | compress | METRIC - time 0.29s
2026-02-09T22:50:55.009253+0900 | compress | METRIC - error 307.88
2026-02-09T22:50:55.010136+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:50:55.010371+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:50:55.011763+0900 | compress_modules | INFO - Quantizing model.layers.11.self_attn.k_proj using 256 samples
2026-02-09T22:50:55.195967+0900 | compress | METRIC - time 0.18s
2026-02-09T22:50:55.196310+0900 | compress | METRIC - error 87.01
2026-02-09T22:50:55.197101+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:50:55.197322+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:50:55.197932+0900 | compress_modules | INFO - Quantizing model.layers.11.self_attn.v_proj using 256 samples
2026-02-09T22:50:55.384542+0900 | compress | METRIC - time 0.19s
2026-02-09T22:50:

(13/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.88it/s]

2026-02-09T22:51:42.908571+0900 | compress_modules | INFO - Quantizing model.layers.12.self_attn.q_proj using 256 samples


2026-02-09T22:51:43.196791+0900 | compress | METRIC - time 0.29s
2026-02-09T22:51:43.197142+0900 | compress | METRIC - error 344.53
2026-02-09T22:51:43.197993+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:51:43.198213+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:51:43.199573+0900 | compress_modules | INFO - Quantizing model.layers.12.self_attn.k_proj using 256 samples
2026-02-09T22:51:43.384945+0900 | compress | METRIC - time 0.19s
2026-02-09T22:51:43.385319+0900 | compress | METRIC - error 94.53
2026-02-09T22:51:43.386124+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:51:43.386324+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:51:43.386953+0900 | compress_modules | INFO - Quantizing model.layers.12.self_attn.v_proj using 256 samples
2026-02-09T22:51:43.570805+0900 | compress | METRIC - time 0.18s
2026-02-09T22:51:

(14/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.87it/s]

2026-02-09T22:52:31.181495+0900 | compress_modules | INFO - Quantizing model.layers.13.self_attn.q_proj using 256 samples


2026-02-09T22:52:31.470868+0900 | compress | METRIC - time 0.29s
2026-02-09T22:52:31.471217+0900 | compress | METRIC - error 386.91
2026-02-09T22:52:31.472020+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:52:31.472218+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:52:31.473561+0900 | compress_modules | INFO - Quantizing model.layers.13.self_attn.k_proj using 256 samples
2026-02-09T22:52:31.656392+0900 | compress | METRIC - time 0.18s
2026-02-09T22:52:31.656757+0900 | compress | METRIC - error 108.32
2026-02-09T22:52:31.657587+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:52:31.657813+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:52:31.658495+0900 | compress_modules | INFO - Quantizing model.layers.13.self_attn.v_proj using 256 samples
2026-02-09T22:52:31.842052+0900 | compress | METRIC - time 0.18s
2026-02-09T22:52

(15/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.89it/s]

2026-02-09T22:53:19.327726+0900 | compress_modules | INFO - Quantizing model.layers.14.self_attn.q_proj using 256 samples


2026-02-09T22:53:19.617019+0900 | compress | METRIC - time 0.29s
2026-02-09T22:53:19.617386+0900 | compress | METRIC - error 419.89
2026-02-09T22:53:19.618215+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:53:19.618421+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:53:19.619928+0900 | compress_modules | INFO - Quantizing model.layers.14.self_attn.k_proj using 256 samples
2026-02-09T22:53:19.815808+0900 | compress | METRIC - time 0.20s
2026-02-09T22:53:19.816158+0900 | compress | METRIC - error 126.15
2026-02-09T22:53:19.816955+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:53:19.817160+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:53:19.817819+0900 | compress_modules | INFO - Quantizing model.layers.14.self_attn.v_proj using 256 samples
2026-02-09T22:53:20.001417+0900 | compress | METRIC - time 0.18s
2026-02-09T22:53

(16/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.78it/s]

2026-02-09T22:54:08.442242+0900 | compress_modules | INFO - Quantizing model.layers.15.self_attn.q_proj using 256 samples


2026-02-09T22:54:08.732526+0900 | compress | METRIC - time 0.29s
2026-02-09T22:54:08.732882+0900 | compress | METRIC - error 435.28
2026-02-09T22:54:08.735541+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:54:08.735770+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:54:08.737196+0900 | compress_modules | INFO - Quantizing model.layers.15.self_attn.k_proj using 256 samples
2026-02-09T22:54:08.920202+0900 | compress | METRIC - time 0.18s
2026-02-09T22:54:08.920560+0900 | compress | METRIC - error 122.69
2026-02-09T22:54:08.921356+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:54:08.921575+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:54:08.922203+0900 | compress_modules | INFO - Quantizing model.layers.15.self_attn.v_proj using 256 samples
2026-02-09T22:54:09.104927+0900 | compress | METRIC - time 0.18s
2026-02-09T22:54

(17/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.82it/s]

2026-02-09T22:54:57.067171+0900 | compress_modules | INFO - Quantizing model.layers.16.self_attn.q_proj using 256 samples


2026-02-09T22:54:57.357630+0900 | compress | METRIC - time 0.29s
2026-02-09T22:54:57.357981+0900 | compress | METRIC - error 515.29
2026-02-09T22:54:57.358821+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:54:57.359019+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:54:57.360363+0900 | compress_modules | INFO - Quantizing model.layers.16.self_attn.k_proj using 256 samples
2026-02-09T22:54:57.545323+0900 | compress | METRIC - time 0.18s
2026-02-09T22:54:57.545674+0900 | compress | METRIC - error 134.84
2026-02-09T22:54:57.546514+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:54:57.546709+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:54:57.547299+0900 | compress_modules | INFO - Quantizing model.layers.16.self_attn.v_proj using 256 samples
2026-02-09T22:54:57.727785+0900 | compress | METRIC - time 0.18s
2026-02-09T22:54

(18/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.79it/s]

2026-02-09T22:55:45.831018+0900 | compress_modules | INFO - Quantizing model.layers.17.self_attn.q_proj using 256 samples


2026-02-09T22:55:46.127265+0900 | compress | METRIC - time 0.30s
2026-02-09T22:55:46.127609+0900 | compress | METRIC - error 532.66
2026-02-09T22:55:46.128456+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:55:46.128655+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:55:46.130162+0900 | compress_modules | INFO - Quantizing model.layers.17.self_attn.k_proj using 256 samples
2026-02-09T22:55:46.312348+0900 | compress | METRIC - time 0.18s
2026-02-09T22:55:46.312791+0900 | compress | METRIC - error 144.69
2026-02-09T22:55:46.313549+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:55:46.313744+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:55:46.314295+0900 | compress_modules | INFO - Quantizing model.layers.17.self_attn.v_proj using 256 samples
2026-02-09T22:55:46.496442+0900 | compress | METRIC - time 0.18s
2026-02-09T22:55

(19/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.80it/s]

2026-02-09T22:56:34.607957+0900 | compress_modules | INFO - Quantizing model.layers.18.self_attn.q_proj using 256 samples


2026-02-09T22:56:34.905943+0900 | compress | METRIC - time 0.30s
2026-02-09T22:56:34.906319+0900 | compress | METRIC - error 585.64
2026-02-09T22:56:34.907194+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:56:34.907420+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:56:34.908732+0900 | compress_modules | INFO - Quantizing model.layers.18.self_attn.k_proj using 256 samples
2026-02-09T22:56:35.100963+0900 | compress | METRIC - time 0.19s
2026-02-09T22:56:35.101308+0900 | compress | METRIC - error 166.38
2026-02-09T22:56:35.102112+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:56:35.102323+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:56:35.102985+0900 | compress_modules | INFO - Quantizing model.layers.18.self_attn.v_proj using 256 samples
2026-02-09T22:56:35.287707+0900 | compress | METRIC - time 0.18s
2026-02-09T22:56

(20/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.81it/s]

2026-02-09T22:57:23.333125+0900 | compress_modules | INFO - Quantizing model.layers.19.self_attn.q_proj using 256 samples


2026-02-09T22:57:23.635533+0900 | compress | METRIC - time 0.30s
2026-02-09T22:57:23.635885+0900 | compress | METRIC - error 588.82
2026-02-09T22:57:23.636924+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:57:23.637153+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:57:23.638484+0900 | compress_modules | INFO - Quantizing model.layers.19.self_attn.k_proj using 256 samples
2026-02-09T22:57:23.831878+0900 | compress | METRIC - time 0.19s
2026-02-09T22:57:23.832220+0900 | compress | METRIC - error 168.03
2026-02-09T22:57:23.832971+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:57:23.833166+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:57:23.833735+0900 | compress_modules | INFO - Quantizing model.layers.19.self_attn.v_proj using 256 samples
2026-02-09T22:57:24.016184+0900 | compress | METRIC - time 0.18s
2026-02-09T22:57

(21/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.84it/s]

2026-02-09T22:58:11.868438+0900 | compress_modules | INFO - Quantizing model.layers.20.self_attn.q_proj using 256 samples


2026-02-09T22:58:12.156826+0900 | compress | METRIC - time 0.29s
2026-02-09T22:58:12.157174+0900 | compress | METRIC - error 698.36
2026-02-09T22:58:12.158874+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:58:12.159087+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:58:12.160366+0900 | compress_modules | INFO - Quantizing model.layers.20.self_attn.k_proj using 256 samples
2026-02-09T22:58:12.341998+0900 | compress | METRIC - time 0.18s
2026-02-09T22:58:12.342331+0900 | compress | METRIC - error 186.94
2026-02-09T22:58:12.343126+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:58:12.343328+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:58:12.343903+0900 | compress_modules | INFO - Quantizing model.layers.20.self_attn.v_proj using 256 samples
2026-02-09T22:58:12.524717+0900 | compress | METRIC - time 0.18s
2026-02-09T22:58

(22/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.84it/s]

2026-02-09T22:59:00.354872+0900 | compress_modules | INFO - Quantizing model.layers.21.self_attn.q_proj using 256 samples


2026-02-09T22:59:00.654832+0900 | compress | METRIC - time 0.30s
2026-02-09T22:59:00.655172+0900 | compress | METRIC - error 801.29
2026-02-09T22:59:00.656826+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:59:00.657047+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:59:00.658301+0900 | compress_modules | INFO - Quantizing model.layers.21.self_attn.k_proj using 256 samples
2026-02-09T22:59:00.840842+0900 | compress | METRIC - time 0.18s
2026-02-09T22:59:00.841182+0900 | compress | METRIC - error 214.45
2026-02-09T22:59:00.841941+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:59:00.842143+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:59:00.842734+0900 | compress_modules | INFO - Quantizing model.layers.21.self_attn.v_proj using 256 samples
2026-02-09T22:59:01.027919+0900 | compress | METRIC - time 0.18s
2026-02-09T22:59

(23/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.85it/s]

2026-02-09T22:59:48.758991+0900 | compress_modules | INFO - Quantizing model.layers.22.self_attn.q_proj using 256 samples


2026-02-09T22:59:49.055358+0900 | compress | METRIC - time 0.30s
2026-02-09T22:59:49.055707+0900 | compress | METRIC - error 874.18
2026-02-09T22:59:49.056515+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:59:49.056735+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T22:59:49.058044+0900 | compress_modules | INFO - Quantizing model.layers.22.self_attn.k_proj using 256 samples
2026-02-09T22:59:49.239729+0900 | compress | METRIC - time 0.18s
2026-02-09T22:59:49.240067+0900 | compress | METRIC - error 247.48
2026-02-09T22:59:49.240854+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T22:59:49.241047+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T22:59:49.241741+0900 | compress_modules | INFO - Quantizing model.layers.22.self_attn.v_proj using 256 samples
2026-02-09T22:59:49.423608+0900 | compress | METRIC - time 0.18s
2026-02-09T22:59

(24/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.82it/s]

2026-02-09T23:00:37.390753+0900 | compress_modules | INFO - Quantizing model.layers.23.self_attn.q_proj using 256 samples


2026-02-09T23:00:37.681118+0900 | compress | METRIC - time 0.29s
2026-02-09T23:00:37.681477+0900 | compress | METRIC - error 973.81
2026-02-09T23:00:37.683061+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T23:00:37.683305+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T23:00:37.684640+0900 | compress_modules | INFO - Quantizing model.layers.23.self_attn.k_proj using 256 samples
2026-02-09T23:00:37.866735+0900 | compress | METRIC - time 0.18s
2026-02-09T23:00:37.867062+0900 | compress | METRIC - error 286.94
2026-02-09T23:00:37.867855+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T23:00:37.868077+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T23:00:37.868668+0900 | compress_modules | INFO - Quantizing model.layers.23.self_attn.v_proj using 256 samples
2026-02-09T23:00:38.053895+0900 | compress | METRIC - time 0.19s
2026-02-09T23:00

(25/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.82it/s]

2026-02-09T23:01:26.005580+0900 | compress_modules | INFO - Quantizing model.layers.24.self_attn.q_proj using 256 samples


2026-02-09T23:01:26.304120+0900 | compress | METRIC - time 0.30s
2026-02-09T23:01:26.304476+0900 | compress | METRIC - error 1390.97
2026-02-09T23:01:26.307311+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T23:01:26.307620+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T23:01:26.309047+0900 | compress_modules | INFO - Quantizing model.layers.24.self_attn.k_proj using 256 samples
2026-02-09T23:01:26.497993+0900 | compress | METRIC - time 0.19s
2026-02-09T23:01:26.498339+0900 | compress | METRIC - error 369.70
2026-02-09T23:01:26.499139+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T23:01:26.499336+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T23:01:26.499946+0900 | compress_modules | INFO - Quantizing model.layers.24.self_attn.v_proj using 256 samples
2026-02-09T23:01:26.683168+0900 | compress | METRIC - time 0.18s
2026-02-09T23:0

(26/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.77it/s]

2026-02-09T23:02:15.096389+0900 | compress_modules | INFO - Quantizing model.layers.25.self_attn.q_proj using 256 samples


2026-02-09T23:02:15.388800+0900 | compress | METRIC - time 0.29s
2026-02-09T23:02:15.389143+0900 | compress | METRIC - error 1585.70
2026-02-09T23:02:15.389984+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T23:02:15.390191+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T23:02:15.391575+0900 | compress_modules | INFO - Quantizing model.layers.25.self_attn.k_proj using 256 samples
2026-02-09T23:02:15.577470+0900 | compress | METRIC - time 0.19s
2026-02-09T23:02:15.577808+0900 | compress | METRIC - error 400.92
2026-02-09T23:02:15.578593+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T23:02:15.578777+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T23:02:15.579525+0900 | compress_modules | INFO - Quantizing model.layers.25.self_attn.v_proj using 256 samples
2026-02-09T23:02:15.767756+0900 | compress | METRIC - time 0.19s
2026-02-09T23:0

(27/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.81it/s]

2026-02-09T23:03:03.786031+0900 | compress_modules | INFO - Quantizing model.layers.26.self_attn.q_proj using 256 samples


2026-02-09T23:03:04.076919+0900 | compress | METRIC - time 0.29s
2026-02-09T23:03:04.077266+0900 | compress | METRIC - error 1896.53
2026-02-09T23:03:04.078096+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T23:03:04.078310+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T23:03:04.079751+0900 | compress_modules | INFO - Quantizing model.layers.26.self_attn.k_proj using 256 samples
2026-02-09T23:03:04.260889+0900 | compress | METRIC - time 0.18s
2026-02-09T23:03:04.261231+0900 | compress | METRIC - error 513.17
2026-02-09T23:03:04.261997+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T23:03:04.262213+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T23:03:04.262826+0900 | compress_modules | INFO - Quantizing model.layers.26.self_attn.v_proj using 256 samples
2026-02-09T23:03:04.444177+0900 | compress | METRIC - time 0.18s
2026-02-09T23:0

(28/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.85it/s]

2026-02-09T23:03:52.313515+0900 | compress_modules | INFO - Quantizing model.layers.27.self_attn.q_proj using 256 samples


2026-02-09T23:03:52.604527+0900 | compress | METRIC - time 0.29s
2026-02-09T23:03:52.604897+0900 | compress | METRIC - error 2861.87
2026-02-09T23:03:52.605733+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T23:03:52.605938+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T23:03:52.607476+0900 | compress_modules | INFO - Quantizing model.layers.27.self_attn.k_proj using 256 samples
2026-02-09T23:03:52.791956+0900 | compress | METRIC - time 0.18s
2026-02-09T23:03:52.792333+0900 | compress | METRIC - error 736.93
2026-02-09T23:03:52.793168+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T23:03:52.793410+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T23:03:52.794176+0900 | compress_modules | INFO - Quantizing model.layers.27.self_attn.v_proj using 256 samples
2026-02-09T23:03:52.983884+0900 | compress | METRIC - time 0.19s
2026-02-09T23:0

(29/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.84it/s]

2026-02-09T23:04:40.816802+0900 | compress_modules | INFO - Quantizing model.layers.28.self_attn.q_proj using 256 samples


2026-02-09T23:04:41.106674+0900 | compress | METRIC - time 0.29s
2026-02-09T23:04:41.107032+0900 | compress | METRIC - error 3286.65
2026-02-09T23:04:41.107876+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T23:04:41.108080+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T23:04:41.109508+0900 | compress_modules | INFO - Quantizing model.layers.28.self_attn.k_proj using 256 samples
2026-02-09T23:04:41.290645+0900 | compress | METRIC - time 0.18s
2026-02-09T23:04:41.291007+0900 | compress | METRIC - error 846.64
2026-02-09T23:04:41.291832+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T23:04:41.292014+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T23:04:41.292722+0900 | compress_modules | INFO - Quantizing model.layers.28.self_attn.v_proj using 256 samples
2026-02-09T23:04:41.472967+0900 | compress | METRIC - time 0.18s
2026-02-09T23:0

(30/31): Calibrating: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:37<00:00,  6.88it/s]

2026-02-09T23:05:29.025362+0900 | compress_modules | INFO - Quantizing model.layers.29.self_attn.q_proj using 256 samples


2026-02-09T23:05:29.313024+0900 | compress | METRIC - time 0.29s
2026-02-09T23:05:29.313369+0900 | compress | METRIC - error 3258.94
2026-02-09T23:05:29.314195+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T23:05:29.314393+0900 | compress | METRIC - Compressed module size: 16.941056 MB
2026-02-09T23:05:29.315786+0900 | compress_modules | INFO - Quantizing model.layers.29.self_attn.k_proj using 256 samples
2026-02-09T23:05:29.499015+0900 | compress | METRIC - time 0.18s
2026-02-09T23:05:29.499354+0900 | compress | METRIC - error 923.58
2026-02-09T23:05:29.500203+0900 | get_GPU_usage_nv | WARNING - Pynml library error:
 NVML Shared Library Not Found
2026-02-09T23:05:29.500419+0900 | compress | METRIC - Compressed module size: 4.235264 MB
2026-02-09T23:05:29.501072+0900 | compress_modules | INFO - Quantizing model.layers.29.self_attn.v_proj using 256 samples
2026-02-09T23:05:29.683879+0900 | compress | METRIC - time 0.18s
2026-02-09T23:0

(31/31): Propagating: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 256/256 [00:00<00:00, 4515.31it/s]

2026-02-09T23:05:40.116133+0900 | finalize | INFO - Compression lifecycle finalized for 1 modifiers
2026-02-09T23:05:40.121411+0900 | post_process | WARNING - Optimized model is not saved. To save, please provide`output_dir` as input arg.Ex. `oneshot(..., output_dir=...)`

[INFO] GPTQ 양자화 완료!
[INFO] vLLM에서 자동으로 Marlin 커널 적용됨


# 6. 모델 저장 및 크기 비교

In [8]:
print("[INFO] 모델 저장 중...")

# 기존 폴더 삭제 후 저장
if os.path.exists(OUT_DIR):
    shutil.rmtree(OUT_DIR)
os.makedirs(OUT_DIR
            , exist_ok=True)

model.save_pretrained(OUT_DIR, save_compressed=True)
tokenizer.save_pretrained(OUT_DIR)

# 파일 확인
print(f"\n[INFO] 저장된 파일:")
total_size = 0
for f in sorted(os.listdir(OUT_DIR)):
    size = os.path.getsize(os.path.join(OUT_DIR, f))
    total_size += size
    print(f"  {f}: {size/1e6:.1f} MB")

quantized_size_gb = total_size / 1e9

print("\n" + "=" * 60)
print("모델 크기 비교")
print("=" * 60)
print(f"  원본 모델:     {ORIGINAL_MODEL_SIZE_GB:.2f} GB")
print(f"  양자화 모델:   {quantized_size_gb:.2f} GB")
print(f"  ----------------------------------------")
print(f"  크기 감소:     {ORIGINAL_MODEL_SIZE_GB - quantized_size_gb:.2f} GB")
print(f"  압축률:        {quantized_size_gb / ORIGINAL_MODEL_SIZE_GB * 100:.1f}%")
print(f"  압축 배수:     {ORIGINAL_MODEL_SIZE_GB / quantized_size_gb:.2f}x")
print("=" * 60)

[INFO] 모델 저장 중...
2026-02-10T00:38:35.879522+0900 | get_model_compressor | INFO - skip_sparsity_compression_stats set to True. Skipping sparsity compression statistic calculations. No sparsity compressor will be applied.


Compressing model: 210it [00:00, 1807.51it/s]



[INFO] 저장된 파일:
  chat_template.jinja: 0.0 MB
  config.json: 0.0 MB
  generation_config.json: 0.0 MB
  merges.txt: 1.2 MB
  model.safetensors: 1407.7 MB
  recipe.yaml: 0.0 MB
  special_tokens_map.json: 0.0 MB
  tokenizer.json: 7.9 MB
  tokenizer_config.json: 0.1 MB
  vocab.json: 1.9 MB

모델 크기 비교
  원본 모델:     2.56 GB
  양자화 모델:   1.42 GB
  ----------------------------------------
  크기 감소:     1.14 GB
  압축률:        55.4%
  압축 배수:     1.80x


# 7. 제출 파일 생성

In [9]:
zip_name = "submit_marlin"
print(f"[INFO] {zip_name}.zip 생성 중...")

# 기존 zip 삭제
if os.path.exists(f"{zip_name}.zip"):
    os.remove(f"{zip_name}.zip")

shutil.make_archive(
    base_name=zip_name,
    format="zip",
    root_dir=".",
    base_dir=OUT_DIR,  # "model" 폴더
)

zip_size = os.path.getsize(f"{zip_name}.zip") / 1e9
print(f"[INFO] 생성 완료: {zip_name}.zip ({zip_size:.2f} GB)")

if zip_size <= 10:
    print("✅ 용량 제한 충족 (≤ 10GB)")
else:
    print("❌ 용량 초과!")

print(f"\n📁 파일 위치: {os.path.abspath(f'{zip_name}.zip')}")
print("\n" + "=" * 60)
print("제출 파일 구조 확인")
print("=" * 60)
print(f"{zip_name}.zip")
print(f"└── model/")
for f in sorted(os.listdir(OUT_DIR)):
    print(f"    ├── {f}")
print("=" * 60)

[INFO] submit_marlin.zip 생성 중...
[INFO] 생성 완료: submit_marlin.zip (0.88 GB)
✅ 용량 제한 충족 (≤ 10GB)

📁 파일 위치: /Users/imdonghyeon/Desktop/lg-aimers8-llm-compression/submit_marlin.zip

제출 파일 구조 확인
submit_marlin.zip
└── model/
    ├── chat_template.jinja
    ├── config.json
    ├── generation_config.json
    ├── merges.txt
    ├── model.safetensors
    ├── recipe.yaml
    ├── special_tokens_map.json
    ├── tokenizer.json
    ├── tokenizer_config.json
    ├── vocab.json


# 8. 예상 성능

### vLLM + Marlin 자동 적용

```
평가 서버 (L4 GPU + vLLM)
         ↓
모델 로드 (GPTQ W4A16, group_size=128)
         ↓
Marlin 커널 자동 감지 및 적용
         ↓
고속 추론 (2~4배 빠름)
```

### 예상 점수

| 항목 | 베이스라인 | Marlin 최적화 |
|------|-----------|---------------|
| PerfNorm | 0.95 | 0.95 (유지) |
| SpeedNorm | 0.30 | 0.50~0.65 |
| **Score** | **0.625** | **0.725~0.80** |

---

## 베이스라인과 차이점

| 항목 | 베이스라인 | Marlin 최적화 |
|------|-----------|---------------|
| group_size | 128 | 128 (동일) |
| actorder | static | **weight** |
| calibration | 256 | 256 (동일) |
| 폴더명 | model_xxx | **model** |

### 핵심 변경
1. `actorder="weight"`: 중요도 기반 가중치 정렬로 정확도 향상
2. `OUT_DIR="./model"`: 제출 규격에 맞는 폴더명

---